btree/btree.c

```diff
@@ -4306,9 +4245,12 @@ int64_t btree_open_tree_tick(struct m0_sm_op *smop)
 		return tree_get(&oi->i_nop, &oi->i_nop.no_addr, P_ACT);
 
 	case P_ACT:
+		if (!oi->i_nop.no_tree->t_type)
```

Comment:
Better to check if (oi->i_nop.no_op.o_sm.sm_rc == 0) as currently code will work, as we always returning -ve values in  case of error condition, But in future we may change op.c which will persist sm_rc and return next state , so in such case you may want to check oi->i_nop.no_op.o_sm.sm_rc in next state i.e. P_ACT. you may refer to P_NEXTDOWN state which is doing same thing. 
Or at least for now, you can have ASSERT(oi->i_nop.no_op.o_sm.sm_rc == 0); it will make sure that tree_get operation was successful.

Reply:
The purpose of this if condition was just to ensure that the tree being returned has tree type intact, but with the new changes it is hard to reach such a scenario.
It wont be replacement actually if we use an sm_rc based check, but that check will be useful.
I will add that.